spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/CloudDataTemplate.java

```diff
@@ -55,10 +60,16 @@ public CloudDataTemplate(URI baseURI) {
 		resources.put("streams/definitions", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/definitions"));
 		resources.put("streams/deployments", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/deployments"));
 		this.streamTemplate = new StreamTemplate(restTemplate, resources);
+		this.counterTemplate = new CounterTemplate(restTemplate, resourceSupport);
```

Comment:
Yes, I wanted to address consistency once we had this discussion.
I think in any case it's better to deal with Links rather than just UriTemplate. The problem with pre-filtering though, is that CloudDataTemplate has knowledge of all the relations, and who is interested in what